pkg/ccl/sqlproxyccl/interceptor/base.go

```diff
@@ -108,7 +108,9 @@ func (p *pgInterceptor) PeekMsg() (typ byte, size int, err error) {
 		return 0, 0, ErrProtocolError
 	}
 
-	return typ, size - 4, nil
+	// Add 1 to size to account for type. We don't need to add 4 (int length) to
+	// it because size is already inclusive of that.
+	return typ, size + 1, nil
```

Comment:
nit: Adding 1 can overflow size into a negative number. change the ErrProtocolError condition to:

if size < 4 || size == math.MaxInt {
  return 0, 0, ErrProtocolError
}